ModulemdTranslationHelpers/Utils.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+import gi
+gi.require_version('Modulemd', '2.0')
+from gi.repository import Modulemd
+
+from babel.messages import Catalog, pofile
+from babel._compat import StringIO, BytesIO
+from datetime import datetime
+
+# Create a Modulemd.ModuleIndex object containing the YAML stream
+fedora_index = Modulemd.ModuleIndex.new()
+fedora_yaml_path = "../modulemd/v2/tests/test_data/f29.yaml"
+ret, failures = fedora_index.update_from_file(fedora_yaml_path, True)
+
+# Retrieve all Modulemd.Module objects
+module_names = fedora_index.get_module_names()
+
+# Create a dictionary where
+# key: stream name
+# value: NSVCA of highest version of that stream
+final_streams = {}
+
+for mod_name in module_names:
+    module = fedora_index.get_module(mod_name)
+
+    # Retrieve all Modulemd.ModuleStreamV2 objects from module
+    streams = module.get_all_streams()
+    for stream in streams:
+        key = stream.get_stream_name()
+        if key not in final_streams.keys():
+            final_streams[key] = stream.get_NSVCA()
+        else:
+            nsvca = stream.get_NSVCA()
+            version = int(nsvca.split(':')[2])
+            if version > int(final_streams[key].split(
+                    ":")[2]):  # found a higher version for same stream
+                final_streams[key] = nsvca
+
+summary_dict = {}
+description_dict = {}
+profile_desc_default_dict = {}
+profile_desc_minimal_dict = {}
+
+for stream_name, nsvca in final_streams.items():
+    mdname = nsvca.split(":")[0]
+    version = int(nsvca.split(":")[2])
+    context = nsvca.split(":")[3]
+    arch = nsvca.split(":")[4]
+    module = fedora_index.get_module(mdname)
+    stream = module.get_stream_by_NSVCA(stream_name, version, context, arch)
+
+    # Process description
+    description = stream.get_description("C")
+    if description not in description_dict.keys():
+        description_dict[description] = [nsvca.rsplit(
+            ":", 3)[0] + ";description"]  # name:stream
+    else:
+        description_dict[description].append(
+            nsvca.rsplit(":", 3)[0] + ";description")
+
+    # Process summary
+    summary = stream.get_summary("C")
+    if summary not in summary_dict.keys():
+        summary_dict[summary] = [
+            nsvca.rsplit(
+                ":",
+                3)[0] +
+            ";summary"]  # name:stream
+    else:
+        summary_dict[summary].append(nsvca.rsplit(":", 3)[0] + ";summary")
+
+    # Process profile descriptions(sometimes NULL)
+    profile_names = stream.get_profile_names()
+    for names in profile_names:
+        profile = stream.get_profile(names)
+        profile_desc = profile.get_description("C")
+        if profile_desc:
+            if profile_desc[0] not in profile_desc_default_dict.keys():
+                profile_desc_default_dict[
+                    profile_desc[0]] = [
+                    nsvca.rsplit(
+                        ":",
+                        3)[0] +
+                    ";profile;default"]
```

Comment:
`default` and `minimal` are not reserved words, they are just examples. The name of a profile can be any string. You need to record that name here. Also, simplify this like I suggested for summary and description.